code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -168,6 +168,13 @@
 		take_damage(30 / severity, BURN, ENERGY, 1)
 	log_message("EMP detected", LOG_MECHA, color="red")
 
+	//Mess with the focus of the inbuilt camera if present
+	if(chassis_camera && !chassis_camera.is_emp_scrambled)
+		chassis_camera.setViewRange(chassis_camera.short_range)
+		chassis_camera.is_emp_scrambled = TRUE
+		diag_hud_set_camera()
+		addtimer(CALLBACK(chassis_camera, TYPE_PROC_REF(/obj/machinery/camera/exosuit, emp_refocus), src), 10 SECONDS / severity)
```

Comment:
Don't make cameras EMP-proof if they're already EMP'd, it's a bit unintuitive

```suggestion
		addtimer(CALLBACK(chassis_camera, TYPE_PROC_REF(/obj/machinery/camera/exosuit, emp_refocus), src), 10 SECONDS / severity, TIMER_OVERRIDE | TIMER_UNIQUE)
```
The above makes it so the timer overrides the previous timer if you call addtimer again while your timer is already ticking, so you don't get emp_refocus called out of sequence.

Reply:
i was actually kind of going for the emp effect's duration to not be refreshed on successful emps, but sure

Reply:
Oh, I mean sure. Can you explain why you think its a good idea? I dont see it

Reply:
continuous EMPs only really come from ion bolts and those are pretty much always just emptied into the mech. this would make the camera stuck to the debuffed view_range for the entire duration of the firefight, which while inconsequential for a carbon mech pilot, it would cripple an AI pilot's FoV completely. having it not refreshed on consecutive EMPs can provide at least some info on the attackers for an AI pilot before getting EMPd and shortsighted again

tl; dr - chaining EMPs to almost completely blind an AI piloted mech is a huge disadvantage. if it's not refreshed on more EMPs, it can level the playing field somewhat better by at least providing a glimpse of what's happening around the mech during an ion/EMP onslaught

Reply:
Okay